### PR TITLE
Dynamic Routes for '/curse/:numeCursa' 

### PR DESCRIPTION
**Pull request by AMihneaa**
The component is designed specifically for desktop. Until it's integrated for mobile as well, it will remain a pull request.

The review is based solely on the design and code implementation for desktop.

### CODE REVIEW

File: client/src/routes/AllRoutes.tsx [hunk 1]
```diff
@@ -26,4 +28,15 @@ export const allRoutes = createBrowserRouter([
...
   },
-]);
+  {
+    path: '/curse',
+    loader: loaderCurse,
+    children: [
+      {
+        path: ":numeCursa?",
+        element: <Curse />,
+
```
Comment (cristim67) [inline client/src/routes/AllRoutes.tsx:38]: remove this enter.

File: client/src/views/Curse.view.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+export function getCurse(): string[]{ 
+    const numeCurs: string[] = curseArr.map(cursa => cursa.numeCurse);
+
+
+    return numeCurs;
+}
+
+const Curse = () => {
+    const {numeCursa} = useParams();
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:108]: add type

File: client/src/views/Curse.view.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    const cursa: CurseValide | undefined = curseArr.find(cursa => cursa.numeCurse === numeCursa);
+
+    if (!cursa) {
+        return <div>Cursa nu a fost găsită!</div>;
+    }
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:114]: !== undefined

File: client/src/views/Curse.view.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+    
+    return (
+        <div>
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:118]: useless div.

File: client/src/routes/AllRoutes.tsx [hunk 5]
```diff
@@ -4,6 +4,8 @@ import About from "../views/About.view";
 import Cause from "../views/Cause.view";
 import Faq from "../views/Faq.view";
 import Contact from "../views/Contact.view";
+import Curse from "../views/Curse.view";
+import { loaderCurse } from "./Loader";
 
 export const allRoutes = createBrowserRouter([
```
Comment (cristim67) [inline client/src/routes/AllRoutes.tsx:10]: type here.

File: client/src/components/Header.component.tsx [hunk 6]
```diff
@@ -56,7 +56,7 @@ const headerTextContent: HeaderItems[] = [
 
 export const Header = () => {
   return (
-    <div className="header h-[5.625rem] w-full fixed top-0 ">
+    <div className="header h-[5.625rem] w-full fixed top-0 z-50">
       <div className="elementHeader h-[80%] my-auto  w-[80rem] mx-auto relative">
         <ul className="flex  p-0 center my-auto h-full">
           {headerTextContent.map((item, index) => (
```
Comment (cristim67) [inline client/src/components/Header.component.tsx:62]: type here

File: client/src/views/Curse.view.tsx (hunk 4 above)
Comment (AMihneaa) [reply client/src/views/Curse.view.tsx:119]: The respective <div> wraps around two components, so it is not useless.

**Comment by AMihneaa** [comment]
**Update !!!**

The component is designed specifically for desktop. Until it's integrated for mobile as well, it will remain a pull request.

The review is based solely on the design and code implementation for desktop.

File: client/src/components/Header.component.tsx [hunk 7]
```diff
@@ -27,40 +30,59 @@ const headerTextContent: HeaderItems[] = [
...
   },
   {
+    id: 5,
     text: "Contacte",
     url: "/contact",
   },
 ];
 
 export const Header = () => {
+  const [ceva, setCeva] = useState(false);
```
Comment (cristim67) [inline client/src/components/Header.component.tsx:65]: use a name that is more suggestive

File: client/src/components/Header.component.tsx [hunk 8]
```diff
@@ -27,40 +30,59 @@ const headerTextContent: HeaderItems[] = [
...
-      1: { text: "Cursa All for One", url: "" },
-      2: { text: "Cursa Copii", url: "" },
-      3: { text: "Feminin 10-16 ani", url: "" },
-      4: { text: "Masculin 10-16 ani", url: "" },
-      5: { text: "Feminim 17-35 de ani", url: "" },
-      6: { text: "Masculin 17-35 de ani", url: "" },
-      7: { text: "Feminin 35 + de ani", url: "" },
-      8: { text: "Masculin 35 + de ani", url: "" },
+      1: { text: "Cursa All for One", url: "/curse/allforone" },
+      2: { text: "Cursa Copii", url: "/curse/kids" },
```
Comment (cristim67) [inline client/src/components/Header.component.tsx:38]: use diacritics

File: client/src/routes/Loader.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
+
+import { ActionFunction, redirect } from "react-router";
+import { getCurse } from "../views/Curse.view";
+
+export const  loaderCurse: ActionFunction  = async ({params}) => {
+    const {numeCursa} = params;
```
Comment (cristim67) [inline client/src/routes/Loader.tsx:6]: type

File: client/src/routes/Loader.tsx [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+
+
+    const numeCurse: string[] = getCurse();
+
+    let isValid = false;
+
+    numeCurse.forEach( curse => {
```
Comment (cristim67) [inline client/src/routes/Loader.tsx:13]: type

File: client/src/views/Curse.view.tsx [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+const curseArr: CurseValide[] = [
+    {
+        numeCurse: "allforone",
+        interfata:{
+            cursa: "All For One",
+            oraStart: "11 : 30",
+            distanta: "400 M",
+            donatieMinima: "OPTIONAL",
+            descriere: "Cursa All For One este o tură populară, necompetitivă, perfectă pentru cei care vor să facă mișcare în natură pe o distanță accesibilă. Aceasta se adresează tuturor persoanelor, de orice vârstă sau gen, chiar și persoanelor cu dizabilități.",
+            urlHarta: "https://www.google.com/maps/embed?pb=!1m42!1m12!1m3!1d1051.7564164316468!2d26.050924936618955!3d44.43943916368305!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m27!3e2!4m5!1s0x40b201c187a44aeb%3A0x776c5045f461c6b0!2sBucure%C5%9Fti%20Sectorul%206%2C%20Bucure%C8%99ti%20061344!3m2!1d44.4396542!2d26.052221199999998!4m3!3m2!1d44.439018999999995!2d26.0515156!4m3!3m2!1d44.4397981!2d26.051081699999997!4m3!3m2!1d44.4400312!2d26.0518355!4m3!3m2!1d44.4399332!2d26.0523311!4m3!3m2!1d44.4396554!2d26.052215!5e0!3m2!1sro!2sro!4v1679949014977!5m2!1sro!2sro",
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:26]: move this in a config file.

File: client/src/components/Header.component.tsx [hunk 12]
```diff
@@ -2,15 +2,18 @@ import { FaFacebook, FaInstagram } from "react-icons/fa";
...
 
 interface HeaderItems {
+  id: number;
   text: string;
   url?: string;
   subText?: Record<number, { text: string; url: string }>;
 }
 
 const headerTextContent: HeaderItems[] = [
   {
```
Comment (cristim67) [inline client/src/components/Header.component.tsx:15]: move this in config file

File: client/src/views/Curse.view.tsx [hunk 13]
```diff
@@ -0,0 +1,134 @@
...
+            cursa: "Feminin 35+ ani",
+            oraStart: "14 : 10",
+            distanta: " 2 KM ",
+            donatieMinima: "40 RON",
+            urlHarta: "https://www.google.com/maps/embed?pb=!1m52!1m12!1m3!1d1359.0739952763695!2d26.051536155519436!3d44.4381565085272!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m37!3e2!4m3!3m2!1d44.4372435!2d26.047853099999998!4m3!3m2!1d44.4392806!2d26.047387!4m3!3m2!1d44.4388363!2d26.0495043!4m3!3m2!1d44.438002399999995!2d26.051835699999998!4m3!3m2!1d44.440619999999996!2d26.0526777!4m3!3m2!1d44.4378836!2d26.054248899999997!4m3!3m2!1d44.4377256!2d26.051095999999998!4m3!3m2!1d44.4380264!2d26.0493903!4m3!3m2!1d44.4372292!2d26.0480509!5e0!3m2!1sro!2sro!4v1679944496994!5m2!1sro!2sro",
+        }
+    },
+];
+
+export function getCurse(): string[]{
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:101]: Fast refresh only works when a file only exports components. Use a new file to share constants or functions between components.

File: client/src/views/Curse.view.tsx [hunk 14]
```diff
@@ -0,0 +1,134 @@
...
+            urlHarta: "https://www.google.com/maps/embed?pb=!1m52!1m12!1m3!1d1359.0739952763695!2d26.051536155519436!3d44.4381565085272!2m3!1f0!2f0!3f0!3m2!1i1024!2i768!4f13.1!4m37!3e2!4m3!3m2!1d44.4372435!2d26.047853099999998!4m3!3m2!1d44.4392806!2d26.047387!4m3!3m2!1d44.4388363!2d26.0495043!4m3!3m2!1d44.438002399999995!2d26.051835699999998!4m3!3m2!1d44.440619999999996!2d26.0526777!4m3!3m2!1d44.4378836!2d26.054248899999997!4m3!3m2!1d44.4377256!2d26.051095999999998!4m3!3m2!1d44.4380264!2d26.0493903!4m3!3m2!1d44.4372292!2d26.0480509!5e0!3m2!1sro!2sro!4v1679944496994!5m2!1sro!2sro",
+        }
+    },
+];
+
+export function getCurse(): string[]{ 
+    const numeCurs: string[] = curseArr.map(cursa => cursa.numeCurse);
+
+
+    return numeCurs;
```
Comment (cristim67) [inline client/src/views/Curse.view.tsx:105]: numeCurs is redundant.

File: client/src/styles/global.tsx [hunk 15]
```diff
@@ -38,6 +38,17 @@ const globalStyles = createGlobalStyle`
...
+  }
+
+  span {
+    color: ${({ theme }: GlobalThemeProps) => theme.text};
+  }
+  p:hover{
+    color: ${({ theme }: GlobalThemeProps) => theme.textHover};
+  }
+
   .header{
```
Comment (cristim67) [inline client/src/styles/global.tsx:52]: this is never used

File: client/src/components/Header.component.tsx (hunk 8 above)
Comment (AMihneaa) [reply client/src/components/Header.component.tsx:38]: no

File: client/src/styles/global.tsx (hunk 15 above)
Comment (AMihneaa) [reply client/src/styles/global.tsx:52]: Take a closer look

**Pull request merged**